galaxy_ng/app/api/ui/viewsets/user.py

```diff
@@ -0,0 +1,49 @@
+from django.shortcuts import get_object_or_404
+
+from rest_framework import mixins
+from rest_framework import permissions as drf_permissions
+
+from galaxy_ng.app.models import auth as auth_models
+from galaxy_ng.app.api import permissions
+from galaxy_ng.app.api.ui import serializers
+from galaxy_ng.app.api import base as api_base
+
+
+class UserViewSet(
+    mixins.ListModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.CreateModelMixin,
+    mixins.UpdateModelMixin,
+    api_base.GenericViewSet,
+):
+
+    def get_serializer_class(self):
+        return serializers.UserSerializer
+
+    def get_queryset(self):
+        return auth_models.User.objects.all()
+
+    def get_permissions(self):
+        permission_list = super().get_permissions()
+        permission_list.append(permissions.IsPartnerEngineer())
```

Comment:
IMPORTNANT: You **must not** modify instance returned by `get_permissions()` method, as it will modify global setting parameter returned by reference from `get_permissions()` method.
```
return super().get_permissions() + [
   permissions.IsPartnerEngineer(),
   permissionsRestrictOnCloudDeployments(),
]
```

Reply:
In that case we need to update the namespace viewsets too

https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/api/ui/viewsets/namespace.py#L45

Reply:
UPD: DRF returns new list object in `get_permissions()` method. So technically it's valid.
However generally it's considered a bad practice to modify objects with lifetime you don't control.